Autopilot/Path Manager/PathManager.c

```diff
@@ -340,7 +340,23 @@ char followWaypoints(PathData* currentWaypoint, float* position, float heading,
 
             *sp_Heading = (int)followStraightPath((float*)&waypointDirection, (float*)targetCoordinates, (float*)position, heading);
         }
-        else{
+        else{            
+            
+            // if target waypoint is a hold waypoint the plane will follow the orbit until the waypoint is updated or removed
+            if (targetWaypoint->type == HOLD_WAYPOINT) {
+                char turnDirection = waypointDirection[0] * nextWaypointDirection[1] - waypointDirection[1] * nextWaypointDirection[0]>0?1:-1;
```

Comment:
This is all repeated code from later lines... It would probably be cleaner if you reused that code, possibly put it in a function, or put the halfPlane stuff in an if statement, in case we have to change this in the future.